### ISSUE TITLE
OauthAutentication doesn't work python-client

### ISSUE THREAD

**Issue by Tlaloc-Es** [apache/pulsar#14885]
**Describe the bug**
Python Oauth2 doesn't work over streamnative cloud, always return 404, but pulsar-client works

**To Reproduce**
1. pipenv install pulsar-client (current 2.9.1)
2. Set your file and configure all as streamnative tell you in their website
```
  params = '''
  {
      "issuer_url": "https://auth.streamnative.cloud/",
      "private_key": "test-test.json",
      "audience": "urn:sn:pulsar:test:test"
  }
```
'''

pulsar_producer = pulsar.Client(
    PULSAR_URL, # pulsar+ssl://test.test.snio.cloud:6651
    authentication=AuthenticationOauth2(params)
)

**Expected behavior**
Pulsar-client connected

**Screenshots**

ERROR [139930200591360] AuthOauth2:223 | Response failed for getting the well-known configuration https://auth.streamnative.cloud/. response Code 404

**Desktop (please complete the following information):**

**Additional context**
The path is correct, I tried with the wrong path and I didn't get errors about the wrong path.
I tried with pulsar-client and works the connection with that file and that configuration.
I am using pipenv with python 3.9.7

**Comment by EronWright** [comment]
The code should strip any trailing slash from the configured `issuerUrl`, and then append `/.well-known/openid-configuration`.  Some issuers have a trailing slash, others don't.  Azure has a wonky issuer URL that doesn't include a trailing slash (and for which normal URL concatenation rules would't work).

**Comment by EronWright** [comment]
I feel this issue should remain open until the Python client has been fixed to trim any trailing slash.  Seems @Tlaloc-Es closed it because they're willing to edit their configuration.

**Comment by BewareMyPower** [comment]
It's fixed by #15928.
